packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -250,17 +250,14 @@ Future<XcodeBuildResult> buildXcodeProject({
     buildCommands.add('-allowProvisioningDeviceRegistration');
   }
 
-  final List<FileSystemEntity> contents = app.project.hostAppRoot.listSync();
-  for (final FileSystemEntity entity in contents) {
-    if (globals.fs.path.extension(entity.path) == '.xcworkspace') {
-      buildCommands.addAll(<String>[
-        '-workspace', globals.fs.path.basename(entity.path),
-        '-scheme', scheme,
-        if (buildAction != XcodeBuildAction.archive) // dSYM files aren't copied to the archive if BUILD_DIR is set.
-          'BUILD_DIR=${globals.fs.path.absolute(getIosBuildDirectory())}',
-      ]);
-      break;
-    }
+  final String? workspacePath = app.project.xcworkspacePath();
+  if (workspacePath != null) {
+    buildCommands.addAll(<String>[
+      '-workspace', globals.fs.path.basename(workspacePath),
```

Comment:
Then here you can get the `.basename` off the `FileSystemEntity` instead of reparsing the string.

Reply:
Updated in https://github.com/flutter/flutter/pull/114099/commits/1596b41ec707513c5f56150bfe088076b71be469.